test/crash_test.dart

```diff
@@ -50,79 +73,158 @@ void main() async {
+          final archiveUrl = Uri.tryParse(pv.archiveUrl);
           if (archiveUrl == null) {
             skipped++;
             return;
           }
           late List<int> archive;
           try {
-            archive = gzip.decode(await c.readBytes(archiveUrl));
+            archive = await c.readBytes(archiveUrl);
           } on http.ClientException {
             skipped++;
             return;
           } on IOException {
             skipped++;
             return;
           }
-          try {
-            await TarReader.forEach(Stream.value(archive), (entry) async {
-              if (entry.name.endsWith('.md')) {
-                late String contents;
-                try {
-                  final bytes = await http.ByteStream(entry.contents).toBytes();
-                  contents = utf8.decode(bytes);
-                } on FormatException {
-                  return; // ignore invalid utf8
-                }
-                try {
-                  markdownToHtml(
-                    contents,
-                    extensionSet: ExtensionSet.gitHubWeb,
-                  );
-                } catch (err, st) {
-                  errors
-                      .add('package:$package/${entry.name}, throws: $err\n$st');
-                }
-              }
-            });
-          } on FormatException {
+
+          final result = await _findMarkdownIssues(
+            pv.package,
+            pv.version,
+            archive,
+          );
+
+          // If tar decoding fails
+          if (result == null) {
             skipped++;
             return;
           }
+
+          errors.addAll(result);
+          result.forEach(print);
         });
         count++;
-        if (DateTime.now().difference(lastStatus) >
-            const Duration(seconds: 30)) {
-          lastStatus = DateTime.now();
-          print('Scanned $count / ${packages.length} (skipped $skipped),'
-              ' found ${errors.length} issues');
-        }
+        status(() =>
+            'Scanned $count / ${packageVersions.length} (skipped $skipped),'
+            ' found ${errors.length} issues');
       }));
 
       await pool.close();
       c.close();
 
+      print('## Finished scanning');
+      print('Scanned ${packageVersions.length} package versions in '
+          '${DateTime.now().difference(started)}');
+
       if (errors.isNotEmpty) {
         print('Found issues:');
         errors.forEach(print);
         fail('Found ${errors.length} cases where markdownToHtml threw!');
       }
     },
-    timeout: const Timeout(Duration(hours: 1)),
+    timeout: const Timeout(Duration(hours: 5)),
     tags: 'crash_test', // skipped by default, see: dart_test.yaml
   );
 }
+
+class PackageVersion {
+  final String package;
+  final String version;
+  final String archiveUrl;
+
+  PackageVersion({
+    required this.package,
+    required this.version,
+    required this.archiveUrl,
+  });
+}
+
+Future<List<String>?> _findMarkdownIssues(
```

Comment:
Could you add a comment here noting what it means when this returns a Future of `null`.